challenge_help_me_carry/src/challenge_hmc_functions/hmc_functions.py

```diff
@@ -0,0 +1,190 @@
+    self.place_position = ds.LockingDesignator(ds.EmptySpotDesignator(robot, self.place_name, 
+                                                                    name="placement",
+                                                                    area=challenge_knowledge.default_area),
+                                                                    name="place_position")
+
+    self.empty_arm_designator = ds.UnoccupiedArmDesignator(robot.arms, robot.rightArm, name="empty_arm_designator")
+
+    # With the empty_arm_designator locked, it will ALWAYS resolve to the same arm (first resolve is cached), unless it is unlocked
+    # For this challenge, unlocking is not needed
+    self.bag_arm_designator = self.empty_arm_designator.lockable()
+    self.bag_arm_designator.lock()
+
+    # We don't actually grab something, so there is no need for an actual thing to grab
+    self.current_item = ds.VariableDesignator(Entity("dummy",
+                                                    "dummy",
+                                                    "/{}/base_link".format(robot.robot_name),
+                                                    kdl_conversions.kdlFrameFromXYZRPY(0.6, 0, 0.5),
+                                                    None,
+                                                    {},
+                                                    [],
+                                                    datetime.datetime.now()),
+                                                    name="current_item")
+
+class WaitForOperatorCommand(smach.State):
+
+    def __init__(self, robot, possible_commands, commands_as_outcomes=False, commands_as_userdata=False):
+
+        self._robot = robot
+        self._possible_commands = possible_commands
+        self._commands_as_outcomes = commands_as_outcomes
+        self._commands_as_userdata = commands_as_userdata
+
+        if commands_as_outcomes:  # each possible command is a separate outcome
+            _outcomes = possible_commands + ['abort']
+        else:                     # outcome is success or abort, recognized command is returned using output_keys
+            _outcomes = ['success', 'abort']
+
+        if commands_as_userdata:  # pass the recognized command to the next state
+            _output_keys = ['command_recognized']
+        else:                     # do not pass data to the next state
+            _output_keys = []
+
+        smach.State.__init__(self, outcomes=_outcomes,  output_keys=_output_keys)
+
+    def _listen_for_commands(self, tries=5, time_out=30):
+        for i in range(0, tries):
+            try:
+                result = self._robot.hmi.query('What command?', 'T -> ' + ' | '.join(self._possible_commands), 'T', timeout=time_out)
+                command_recognized = result.sentence
+            except TimeoutException:
+                command_recognized = None
+            if command_recognized == "":
```

Comment:
I'd expect an empty line before the 'if', as the previous code was about getting a command, and now you do the next step by processing what you received.

Is it by design that you use both `""` and `None` as values? If you you may want to document what each value means.